### PR TITLE
Dr add from chat

### LINKED ISSUE #19: Friends - Add from chat
Given an active user sees a chat message from another user
And wants to make that user a friend
When the active user clicks on the other user's name in the chat history
Then a message should be presented to the active user asking for verification of adding the other user to their friend list

### PR DESCRIPTION
**Pull request by Daniel-L-Ross**
# Description

Added dynamic link to each name on the chat. 
The link takes the user to a confirm add friend page.
If the user=== current user or if the user is already friends, the user cannot add the friend.
Otherwise the user can confirm or go back to messages.

Closes #19

## Type of change

- [ ] New feature

## Testing

- run a json-server

```
git fetch (branch-name)
npm start
```

- [x] Click on a user you are not friends with in chat then add the user
- [x] Click on a user you are friends with
- [x] Click on your own name in chat

### CODE REVIEW

File: src/components/messages/MessageCard.js [hunk 1]
```diff
@@ -26,7 +27,11 @@ export const MessageCard = ({ message, user, currentUser }) => {
     }
     return (
         <article className="message message--card">
-            <div className="message_text"><em>{user.name}:</em> {message.text}</div>
+            <div className="message_text">
+                <Link to={`/friends/add/${user.id}`}>
+                    <em>{user.name}:</em>
```
Comment (SubtleCo) [inline src/components/messages/MessageCard.js:32]: Tiny request: add a space after the colon in the <em> tags to make it a little more readable?

File: src/components/users/UserList.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const UserList = () => {
         let isFriends = false
         return users.filter(user => {
             const friendCheck = filteredFriends.find(f => f.userId === user.id)
-            if ( friendCheck !== undefined) {
+            if (friendCheck !== undefined) {
```
Comment (SubtleCo) [inline src/components/users/UserList.js:33]: What we talked about - friendCheck should be responsible for friend data on its own

**Review by SubtleCo — CHANGES_REQUESTED** [review]

**Review by SubtleCo — APPROVED** [review]
LOVE it

**Review by SLLittrell — APPROVED** [review]
Looks Good Dan!!

**Pull request merged**